pkg/hyper/network.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hyper
+
+import (
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/containernetworking/cni/libcni"
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	DefaultInterfaceName = "eth0"
+	DefaultNetDir        = "/etc/cni/net.d"
+	DefaultCNIDir        = "/opt/cni/bin"
+)
+
+type InterfaceInfo struct {
+	Ip string
+}
+
+type CNINetworkPlugin struct {
+	sandboxNs map[string]*sandboxNetNS
+}
+
+func InitCNI() (*CNINetworkPlugin, error) {
+	plugin := &CNINetworkPlugin{}
+
+	plugin.sandboxNs = make(map[string]*sandboxNetNS)
```

Comment:
Read and check network config here? So we could fail fast and don't need to read and check again and again later.